.github/workflows/qe-image.yml

```diff
@@ -0,0 +1,57 @@
+name: Push qe images for new tag
+on:
+  push:
+    tags:
+      - '*'
+env:
+  IMAGE_REGISTRY: quay.io
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: 
+          - ubuntu-latest
+        go: 
+          - '1.21'    
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v4
+      - name: Log in to Quay.io
+        uses: redhat-actions/podman-login@v1
+        with:
+          username: ${{ secrets.QUAY_IO_USERNAME }}
+          password: ${{ secrets.QUAY_IO_PASSWORD }}
+          registry: ${{ env.IMAGE_REGISTRY }}
+      - name: create script of building and push images
+        run: |
+          cat <<EOF > build-qe-images.sh
+          !#/bin/bash
+          set -x
+          OS=linux ARCH=amd64 CRC_INTEGRATION_IMG_VERSION=\${1} make containerized_integration
+          podman push quay.io/crcont/crc-integration:\${1}-linux-amd64
+          OS=windows ARCH=amd64 CRC_INTEGRATION_IMG_VERSION=\${1} make containerized_integration
+          podman push quay.io/crcont/crc-integration:\${1}-windows-amd64
+          OS=darwin ARCH=amd64 CRC_INTEGRATION_IMG_VERSION=\${1} make containerized_integration
+          podman push quay.io/crcont/crc-integration:\${1}-darwin-amd64
+          OS=darwin ARCH=arm64 CRC_INTEGRATION_IMG_VERSION=\${1} make containerized_integration
+          podman push quay.io/crcont/crc-integration:\${1}-darwin-arm64
+          OS=linux ARCH=amd64 CRC_E2E_IMG_VERSION=\${1} make containerized_e2e
+          podman push quay.io/crcont/crc-e2e:\${1}-linux-amd64
+          OS=windows ARCH=amd64 CRC_E2E_IMG_VERSION=\${1} make containerized_e2e
+          podman push quay.io/crcont/crc-e2e:\${1}-windows-amd64
+          OS=darwin ARCH=amd64 CRC_E2E_IMG_VERSION=\${1} make containerized_e2e
+          podman push quay.io/crcont/crc-e2e:\${1}-darwin-amd64
+          OS=darwin ARCH=arm64 CRC_E2E_IMG_VERSION=\${1} make containerized_e2e
+          podman push quay.io/crcont/crc-e2e:\${1}-darwin-arm64
```

Comment:
Instead of pushing each image one by one, can you have a manifest and add those images to manifest and then push that manifest against version (something like we do for bundle images https://quay.io/repository/crcont/podman-bundle?tab=tags)?

Reply:
@praveenkumar , Thanks for this suggestion. Manifest is a great way to manage the images. 
But I failed to pull a manifest image without specifying the os and arch:
```
% podman pull [quay.io/crcont/podman-bundle:4.4.4](http://quay.io/crcont/podman-bundle:4.4.4)
Trying to pull [quay.io/crcont/podman-bundle:4.4.4](http://quay.io/crcont/podman-bundle:4.4.4)...
Error: choosing an image from manifest list docker://quay.io/crcont/podman-bundle:4.4.4: no image found in image index for architecture arm64, variant "v8", OS linux
```
I know can use `--platform` to get a specific one. But there is no option in tekton definition to specify the os and arch.

Reply:
The strategy here is not possible, because the actual platform for all the images is linux / amd64 the suffix on the versioning is due to the content the have inside (test binary for X platform)

These images are executed locally (i.e. gh runner, or pod on ocp) and they connect against a remote target, copy resources to it, execute them and finally copy back the results)

Reply:
Actually I am thinking on https://github.com/adrianriobo/deliverest/issues/44 because most of the size of the image is due to the base image and no the binary they hold inside so we are not getting much benefit from current split per os and arch.